src/segmantic/seg/dataset.py

```diff
@@ -107,14 +112,74 @@ def check_matching_filenames(self):
                     f"The pair image/label pair {d['image']} : {d['label']} doesn't correspond."
                 )
 
-    def dump_dataset(self) -> str:
-        return json.dumps(
-            {
-                "training": self._train_files + self._val_files,
-                "split": [len(self._train_files), len(self._val_files)],
-            },
-            cls=PathEncoder,
-        )
+    def dump_dataset(self, cross_val: bool = False) -> str:
```

Comment:
I think I understand why you added a different format for saving the dataset when called from cross-validation case. This information would be stored in the `split: [num_training, num_validation]` property. But so far we don't read it. Could you not get rid of this if/else branching and the `load_from_cross_val_json` function below, if we would actually read and use the split properly in `load_from_json`?

Reply:
implemented as discussed this morning~~